cmd/sovereignnode/incomingHeader/incomingHeaderProcessor_test.go

```diff
@@ -71,8 +78,82 @@ func createIncomingHeadersWithIncrementalRound(numRounds uint64) []sovereign.Inc
 	return ret
 }
 
+func createTokenData(amount *big.Int) []byte {
```

Comment:
All of these helper functions are **not ok**.
1. You are creating new args in each one of them. If I call these with my custom args, these are not compatible.
2. You create whole structs just to use the data codec
3. All of these **shall be removed** once you use real mocks/stubs, since you won't need them, you will define the expected result in the test.